google_takeout_parser/locales/common.py

```diff
@@ -0,0 +1,27 @@
+"""
+This file defines general takout-files.
+locale HandlerMaps should match strings against these files to allow auto-parser selection.
+Parsers are mapped in google_takeout_parser/path_handler.py
+"""
+
+from enum import Enum, auto
+from typing import (
+    Dict,
+    Optional
+)
+
+class TakeoutFile(Enum):
+    CHROME_HISTORY = auto(),
+    GPLAYSTORE_INSTALLS = auto(),
+    LOCATION_HISTORY = auto(),
+    LOCATION_HISTORY_SEMANTIC = auto(),
+    YOUTUBE_HISTORY_HTML = auto(),
+    YOUTUBE_HISTORY_JSON = auto(),
+    YOUTUBE_COMMENT = auto(),
+    YOUTUBE_COMMENT_LIVECHAT = auto(),
+    YOUTUBE_LIKES = auto(),
+    # Parent Folder "My Activity"
+    ACTIVITY_HTML = auto(),
+    ACTIVITY_JSON = auto()
+
+HandlerMap = Dict[str, Optional[TakeoutFile]]
```

Comment:
Hmm, I sort of see this enum adding more problems than it solves. It means the user has to redefine/subclass this enum if they ever want to add a custom file this parses without just forking and editing this directly, and they also have to edit the global variable `TAKEOUT_PARSER` to add their function there

It does mean you have to import the functions at the top of every locale file, but I think I would actually prefer that.

Smaller issue is that this adds indirection in that you have to know about `.common.py` and the `TAKEOUT_PARSER` map. If someone was exploring this and the functions were inlined here, its much easier to jump from `locales/de.py` to the `_parse_json_activity` function than having to explore to make sure you're going to the correct `Enum` value from the `TAKEOUT_PARSER` dict

There is a bit of personal bias here in that `Enum`s are not as powerful in python than in other languages so unless I have a good reason to use them I won't, but unless you have some reason not to I think it'd be better to just put the functions directly in the handler map like they were before

Reply:
After some time thinking about it I generally agree with mapping to functions directly.  

All in the layer of indirection only allows changing parser-functions for all locales at once using the `ParserMap`.
When adding a new locale one would also look at `en.py` anyway. Doesn't matter if you copy functions or Enums from there.

I still want to present my thought-process so we're on the same page.  

If you still agree on including functions directly into localized maps I'll do it, leaving only type definitions and the `LocalizedHandler` handler class in `locale/common.py`.

also there's actually a circular import in the `locale`-Files:
- common.py imports all langs to provide them in `LocalizedHandler`
- each lang imports common.py to get the typedef `HandlerMap`

***Is this a problem when working with `from x import y`?***

### my thought process

In my opinion a localized handler map should only account for the localization-differences, in this case meaning   
  
1. A HandlerMap maps the locale name to a **general file name**, currently implemented as the Enum `TakeoutFile`
2. The ParserMap [TAKEOUT_PARSER](https://github.com/parthux1/google_takeout_parser/blob/5d2c905afba3718e5858632398d95692bd2f031b/google_takeout_parser/path_handler.py#L35) (should definetly rename it. TakeoutParser is already a class) knows how to parse **general file names**

Using this concept each locale would have to map their custom regexes to a "standardized" file type.

### Mentioned problem: adding custom parsers
> It means the user has to redefine/subclass this enum if they ever want to add a custom file this parses without just forking and editing this directly, and they also have to edit the global variable TAKEOUT_PARSER to add their function there

The current implementation will try to parse a file by iterating over all parsers.
[dispatch_map](https://github.com/parthux1/google_takeout_parser/blob/5d2c905afba3718e5858632398d95692bd2f031b/google_takeout_parser/path_dispatch.py#L140) allow passing functions.
By supplying a custom map you could add custom parsers:
1. Put all own parsers, as well as their "trigger-regex" in a custom `HandlerMap`
2. Define a list containing the custom `handlerMap` before any standard-HandlerMap
4. Supply this list to `TakeoutParser`
```python
my_map: HandlerMap = {r"regex/to/file": my_handler}
t = TakeoutParser("path/to/takeout", handlers=[my_map, LocalizedHandler.EN()])
```

Reply:
Regarding the circular import, you could instead move the `LocalizedHandler` which collects all the maps to something like an `locales/all.py` file, and only leave the types/shared functions in `common.py`. 

> custom parsers

Oh, yeah, if the enum is removed there is no issue with providing a custom parser, your though process makes sense there.